src/Language/PureScript/TypeChecker/Subsumption.hs

```diff
@@ -102,29 +98,12 @@ subsumes' SElaborate (ConstrainedType _ con ty1) ty2 = do
   elaborate <- subsumes' SElaborate ty1 ty2
   let addDicts val = App val (TypeClassDictionary con dicts hints)
   return (elaborate . addDicts)
-subsumes' mode (TypeApp _ f1 r1) (TypeApp _ f2 r2) | eqType f1 tyRecord && eqType f2 tyRecord = do
-    let goWithLabel l t1 t2 = withErrorMessageHint (ErrorInRowLabel l) $ subsumes' SNoElaborate t1 t2
-    let (common, ((ts1', r1'), (ts2', r2'))) = alignRowsWith goWithLabel r1 r2
-    -- For { ts1 | r1 } to subsume { ts2 | r2 } when r1 is empty (= we're working with a closed row),
-    -- every property in ts2 must appear in ts1. If not, then the candidate expression is missing a required property.
-    -- Conversely, when r2 is empty, every property in ts1 must appear in ts2, or else the expression has
-    -- an additional property which is not allowed.
-    when (isREmpty r1')
-      (for_ (firstMissingProp ts2' ts1') (throwError . errorMessage . PropertyIsMissing . rowListLabel))
-    when (isREmpty r2')
-      (for_ (firstMissingProp ts1' ts2') (throwError . errorMessage . AdditionalProperty . rowListLabel))
-    -- Check subsumption for common labels
-    sequence_ common
-    -- Inject the info here
-    unifyTypes (rowFromList (ts1', r1')) (rowFromList (ts2', r2'))
-    -- Nothing was elaborated, return the default coercion
-    return (defaultCoercion mode)
-  where
-    -- Find the first property that's in the first list (of tuples) but not in the second
-    firstMissingProp t1 t2 = fst <$> uncons (minusBy' (comparing rowListLabel) t1 t2)
-subsumes' mode ty1 ty2@(TypeApp _ obj _) | obj == tyRecord =
-  subsumes' mode ty2 ty1
```

Comment:
I'm guessing this case was removed so that the `isOrdered` part of `TypesDoNotUnify` is correct? Are there any other ramifications here?

Reply:
Yeah, that's right. It was a remarkably useless clause anyway, assuming I didn't miss anything; going through the other patterns matched previously, I think it's impossible for `ty1` here to be any type that would trigger a previous clause with a `TypeApp` in the first position. In which case, this clause was just an overly complicated way to fall through to the final clause with the types in the wrong order.